pgmpy/models/continuous/JointGaussianDistribution.py

```diff
@@ -0,0 +1,229 @@
+                else would return a new distribution.
+
+        Returns
+        -------
+        JointGaussianDistribution or None :
+                if inplace=True (default) returns None
+                if inplace=False return a new JointGaussianDistribution instance
+
+        Examples
+        --------
+        >>> from pgmpy.models import JointGaussianDistribution as JGD
+        >>> dis = JGD(['x1', 'x2', 'x3'], np.array([[1], [-3], [4]]),
+                        np.array([[4, 2, -2], [2, 5, -5], [-2, -5, 8]]))
+        >>> dis.variables
+        ['x1', 'x2', 'x3']
+        >>> dis.mean_vector
+        matrix([[ 1],
+                [-3],
+                [ 4]])
+        >>> dis.covariance_matrix
+        matrix([[ 4,  2, -2],
+                [ 2,  5, -5],
+                [-2, -5,  8]])
+
+        >>> dis.marginalize(['x3'])
+        dis.variables
+        ['x1', 'x2']
+        >>> dis.mean_vector
+        matrix([[ 1],
+                [-3]]))
+        >>> dis.covariance_matrix
+        np.matrix([[4, 2],
+                   [2, 5]])
+        """
+
+        if isinstance(variables, six.string_types):
+            raise TypeError("variables: Expected type list or array-like, got type str")
+
+        distribution = self if inplace else self.copy()
+
+        var_indexes = [distribution.variables.index(var) for var in variables]
+        index_to_keep = sorted(set(range(len(self.variables))) - set(var_indexes))
+
+        distribution.variables = [distribution.variables[index] for index in index_to_keep]
+        distribution.mean_vector = distribution.mean_vector[index_to_keep]
+        distribution.covariance_matrix = distribution.covariance_matrix[np.ix_(index_to_keep, index_to_keep)]
+        distribution._precision_matrix = None
+
+        if not inplace:
+            return distribution
+
+    def copy(self):
+        """
+        Return a copy of the distribution.
+
+        Returns
+        -------
+        JointGaussianDistribution: copy of the distribution
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from pgmpy.models import JointGaussianDistribution as JGD
+        >>> gauss_dis = JGD(['x1', 'x2', 'x3'], np.array([[1], [-3], [4]]),
+                            np.array([[4, 2, -2], [2, 5, -5], [-2, -5, 8]]))
+        >>> copy_dis = gauss_dis.copy()
+        >>> copy_dis.variables
+        ['x1', 'x2', 'x3']
+        >>> copy_dis.mean_vector
+        matrix([[ 1],
+                [-3],
+                [ 4]])
+        >>> copy_dis.covariance_matrix
+        matrix([[ 4,  2, -2],
+                [ 2,  5, -5],
+                [-2, -5,  8]])
+        >>> copy_dis.precision_matrix
+        matrix([[ 0.3125    , -0.125     ,  0.        ],
+                [-0.125     ,  0.58333333,  0.33333333],
+                [ 0.        ,  0.33333333,  0.33333333]])
+        """
+        copy_distribution = JointGaussianDistribution(self.variables, self.mean_vector,
```

Comment:
@khalibartan It works fine because in the __init__ method the np.matrix command creates a copy.
```python
>>> a = np.matrix([[1,2,3,4],[0,0,0,0]])
a
matrix([[1, 2, 3, 4],
        [0, 0, 0, 0]])
>>> b = a
>>> a[1,1] = 99999
>>> a
matrix([[    1,     2,     3,     4],
        [    0, 99999,     0,     0]])
>>> b
matrix([[    1,     2,     3,     4],
        [    0, 99999,     0,     0]])

>>> a = np.matrix([[1,2,3,4],[0,0,0,0]])
>>> b = np.matrix(a)
>>> a[1,1] = 99999
>>> a
matrix([[    1,     2,     3,     4],
        [    0, 99999,     0,     0]])
>>> b
matrix([[1, 2, 3, 4],
       [0, 0, 0, 0]])
```